hudi-common/src/main/java/org/apache/hudi/common/table/read/BufferedRecordMergerFactory.java

```diff
@@ -161,7 +164,21 @@ public Pair<Boolean, T> finalMerge(BufferedRecord<T> olderRecord,
    * An implementation of {@link BufferedRecordMerger} which merges {@link BufferedRecord}s
    * based on {@code EVENT_TIME_ORDERING} merge mode.
    */
-  private static class EventTimeBufferedRecordMerger<T> implements BufferedRecordMerger<T> {
+  static class EventTimeBufferedRecordMerger<T> implements BufferedRecordMerger<T> {
```

Comment:
Add a new `BufferedRecordMerger` to avoid unnecessary cost for row-level option `hoodie.write.track.event.time.watermark` check.